src/components/Items/Item.tsx

```diff
@@ -62,10 +63,23 @@ const Item: React.FC<ItemProps> = ({ index, item }) => {
   const onFocus = () => setIsFocused(true);
   const onBlur = () => setIsFocused(false);
 
+  /**
+   * if a handleItemInteraction() callback is present
+   * pass `item` object up to consuming application.
+   * if not, allow browser to route user to href value.
+   */
+  const handleAnchorClick = (e: MouseEvent<HTMLAnchorElement>) => {
+    if (!handleItemInteraction) return;
+
+    e.preventDefault();
+    handleItemInteraction(item);
```

Comment:
My first thought was this is heavy, yet convenient to pass back the whole `item` object as opposed to an item `id` and make the consumer hunt for it.  Good call.